perfkitbenchmarker/linux_packages/maven.py

```diff
@@ -16,10 +16,14 @@
 import posixpath
 
 from perfkitbenchmarker import flags
+from perfkitbenchmarker import data
 from perfkitbenchmarker.linux_packages import INSTALL_DIR
 from six.moves.urllib.parse import urlparse
 
-flags.DEFINE_string('maven_version', '3.6.3', 'The version of maven')
+flags.DEFINE_string('maven_version', '3.6.3',
+                    'The version of maven')
+flags.DEFINE_string('maven_mirror_url', '',
```

Comment:
Why empty string instead of None as the default?

Reply:
Also, @yuyantingzero can/should we require this to be https for security?

Reply:
@dlott good point. I have made the default to be `None`.

As far as requiring https, I see that all the mirrors specified in the [maven documentation](https://repo.maven.apache.org/maven2/.meta/repository-metadata.xml) are actually http. Mirrors are also often used internally, on private networks, and setting something like this up with https would probably be a bit complicated.

Reply:
HTTPs is preferred, but IIRC, some packages is only available from HTTP mirror.

One hack I did when not able of find HTTPs source is building from source code directly.

Correct me if I am wrong, maven is available from github: https://github.com/apache/maven/tree/maven-3.6.3. 
You can potentially skip the compilation and download the binary zip directly from github. https://github.com/apache/maven/archive/maven-3.6.3.zip

Reply:
@yuyantingzero the `maven_mirror_url` does not refer to a location for downloading maven, but for downloading certain Maven artifacts and plugins, as specified by the target project (being built using Maven) in their pom.xml file. This is a Maven setting, and more details on Maven mirrors can be found [here](http://maven.apache.org/guides/mini/guide-mirror-settings.html).

Maven has a default mirror from which it downloads these artifacts/plugins, but sometimes the user might want to use a different (internal) mirror, to speed things up, hence this flag.

Please let me know if you have additional changes and if I should change the name of the flag to be more descriptive.

Reply:
I see. Thanks for the explanation.
Looks good to me.
If user choose a non-default mirror, it is their responsibility to ensure security.

Reply:
Sounds good, thanks for the feedback.